### PR TITLE
Fix bug causing crier to not update.

### PR DESCRIPTION
**Pull request by spxtr**
Along with a test that fails before and passes after.

### CODE REVIEW

**Comment by spxtr** [comment]
ping @rmmh

File: prow/crier/client_test.go [hunk 1]
```diff
@@ -54,9 +55,9 @@ func (f *fakeGitHub) CreateComment(org, repo string, number int, comment string)
 }
 
 func (f *fakeGitHub) EditComment(org, repo string, ID int, comment string) error {
-	for _, ic := range f.ics {
```
Comment (rmmh) [inline prow/crier/client_test.go]: oh right, range is done by value :)

File: prow/crier/crier.go [hunk 2]
```diff
@@ -265,7 +265,7 @@ func (s *Server) handle(r Report) error {
 			return fmt.Errorf("error creating comment: %v", err)
 		}
 	} else if len(entries) > 0 {
-		if err := s.ghc.EditComment(r.RepoOwner, r.RepoName, r.Number, createComment(r, entries)); err != nil {
```
Comment (rmmh) [inline prow/crier/crier.go]: we're storing comments by their ID for the whole repo, not against specific issues?

Comment (spxtr) [inline prow/crier/crier.go]: That's how GitHub stores them, yes. Kinda strange.

**Comment by spxtr** [comment]
@rmmh ptal

**Pull request merged**